src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Windows.cs

```diff
@@ -0,0 +1,562 @@
+                    }
+                }
+            }
+            catch
+            {
+                vts.Dispose();
+                throw;
+            }
+
+            // Completion handled by callback.
+            vts.FinishedScheduling();
+            return new ValueTask<int>(vts, vts.Version);
+        }
+
+        private static async ValueTask<long> ReadScatterAtOffsetMultipleSyscallsAsync(SafeFileHandle handle, IReadOnlyList<Memory<byte>> buffers,
+            long fileOffset, CancellationToken cancellationToken)
+        {
+            long total = 0;
+
+            for (int i = 0; i < buffers.Count; i++)
+            {
+                Memory<byte> buffer = buffers[i];
+                int read = await ReadAtOffsetAsync(handle, buffer, fileOffset + total, cancellationToken).ConfigureAwait(false);
+                total += read;
+
+                if (read != buffer.Length)
+                {
+                    break;
+                }
+            }
+
+            return total;
+        }
+
+        private static ValueTask<long> WriteGatherAtOffsetAsync(SafeFileHandle handle, IReadOnlyList<ReadOnlyMemory<byte>> buffers,
+            long fileOffset, CancellationToken cancellationToken)
+        {
+            if (CanUseScatterGatherWindowsAPIs(handle))
+            {
+                long totalBytes = 0;
+                for (int i = 0; i < buffers.Count; i++)
+                {
+                    totalBytes += buffers[i].Length;
+                }
+
+                if (totalBytes <= int.MaxValue) // the ReadFileScatter API uses int, not long
+                {
+                    return WriteGatherAtOffsetSingleSyscallAsync(handle, buffers, fileOffset, (int)totalBytes, cancellationToken);
+                }
+            }
+
+            return WriteGatherAtOffsetMultipleSyscallsAsync(handle, buffers, fileOffset, cancellationToken);
+        }
+
+        private static async ValueTask<long> WriteGatherAtOffsetMultipleSyscallsAsync(SafeFileHandle handle, IReadOnlyList<ReadOnlyMemory<byte>> buffers,
+            long fileOffset, CancellationToken cancellationToken)
+        {
+            long total = 0;
+
+            for (int i = 0; i < buffers.Count; i++)
+            {
+                ReadOnlyMemory<byte> buffer = buffers[i];
+                int written = await WriteAtOffsetAsync(handle, buffer, fileOffset + total, cancellationToken).ConfigureAwait(false);
+                total += written;
+
+                if (written != buffer.Length)
+                {
+                    break;
+                }
+            }
+
+            return total;
+        }
+
+        private static async ValueTask<long> WriteGatherAtOffsetSingleSyscallAsync(SafeFileHandle handle, IReadOnlyList<ReadOnlyMemory<byte>> buffers,
+            long fileOffset, int totalBytes, CancellationToken cancellationToken)
+        {
+            if (buffers.Count == 1)
+            {
```

Comment:
Nit: missing the same comment you have at https://github.com/dotnet/runtime/pull/53669/files#diff-8e1fc498b4c506832ef1f17dc9eec9324657413163179b809f45b82830307182R325

Reply:
I usually comment given pattern only once, at the first occurrence

Reply:
Ok... but you, for example, copied the comment a few lines below in both places:
https://github.com/dotnet/runtime/pull/53669/files/c87354c999fa4400956ff2cb69e92e82ec41e931#diff-8e1fc498b4c506832ef1f17dc9eec9324657413163179b809f45b82830307182R478